backtracking/subarray_sum.cpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file
+ * @brief We are given with an array and a sum value. The algorithms find all
+ * the subarrays of that array with sum equal to given sum and return such
+ * subarrays count. This approach will have \f$O(n)\f$ time complexity and
+ * \f$O(n)\f$ space complexity. NOTE: In this problem, we are only refering to
+ * the continuous subsets as subarrays everywhere. Subarrays can be created
+ * using deletion operation at the end or the front of an array only. The parent
+ * array is also counted in subarrays having 0 number of deletion operations.
+ * @details Subset sum(only continuous subsets) problem
+ * (https://en.wikipedia.org/wiki/Subset_sum_problem)
+ * @author [Swastika Gupta](https://github.com/Swastyy)
+ */
+
+#include <cassert>        /// for assert
+#include <iostream>       /// for IO operations
+#include <unordered_map>  /// for unordered_map
+#include <vector>         /// for std::vector
+
+/**
+ * @namespace backtracking
+ * @brief subarray sum algorithm
+ */
+namespace backtracking {
+/**
+ * @namespace subarraySum
+ * @brief Functions for the [Subset
+ * sum](https://en.wikipedia.org/wiki/Subset_sum_problem) implementation
+ */
+namespace subarraySum {
+/**
+ * @brief The main function that implements the count of the subarrays
+ * @param sum is the required sum of any subarrays
+ * @param in_arr is the input array
+ * @returns count of the number of subsets with required sum
+ */
+
+uint64_t subarray_sum(int sum, const std::vector<int> &in_arr) {
+    int nelement = in_arr.size();
```

Comment:
Remember to use `uint64` for non-negative values, `int64_t` for negative values, or their appropriate size [reference](https://en.cppreference.com/w/cpp/types/integer).

Reply:
Ohh okk!